pallets/connectors/src/lib.rs

```diff
@@ -224,36 +226,54 @@ pub mod pallet {
 	#[pallet::error]
 	pub enum Error<T> {
 		/// Failed to map the asset to the corresponding Connector's General
-		/// Index representation
+		/// Index representation and thus cannot be used as an
+		/// ivnestment currency.
 		AssetNotFound,
-		/// A pool could not be found
+		/// The metadata of the given asset does not declare it as a pool
+		/// currency and thus it cannot be used as an investment currency.
+		AssetMetadataNotPoolCurrency,
+		/// The given asset does not match the currency of the pool.
+		AssetNotPoolCurrency,
+		/// A pool could not be found.
 		PoolNotFound,
-		/// A tranche could not be found
+		/// A tranche could not be found.
 		TrancheNotFound,
-		/// Could not find the metadata of a tranche token
+		/// Could not find the metadata of a tranche token.
 		TrancheMetadataNotFound,
 		/// Failed to fetch a tranche token price.
 		/// This can occur if `TrancheNotFound` or if effectively
 		/// the price for this tranche has not yet been set.
 		MissingTranchePrice,
-		/// Router not set for a given domain
+		/// Router not set for a given domain.
 		MissingRouter,
-		/// Transfer amount must be non-zero
+		/// Transfer amount must be non-zero.
 		InvalidTransferAmount,
-		/// A transfer to a non-whitelisted destination was attempted
+		/// A transfer to a non-whitelisted destination was attempted.
 		UnauthorizedTransfer,
-		/// Failed to build Ethereum_Xcm call
+		/// Failed to build Ethereum_Xcm call.
 		FailedToBuildEthereumXcmCall,
-		/// The origin of an incoming message is not in the allow-list
+		/// The origin of an incoming message is not in the allow-list.
 		InvalidIncomingMessageOrigin,
-		/// Failed to decode an incoming message
+		/// Failed to decode an incoming message.
 		InvalidIncomingMessage,
-		/// A transfer attempt from the local to the local domain
-		InvalidTransferDomain,
+		/// The destination domain is invalid.
+		InvalidDomain,
+		/// The validity is in the past.
+		InvalidValidity,
```

Comment:
Maybe we should make this error name a bit more specific

Reply:
Do you think `InvalidtTrancheInvestorValidity` suffices?

Reply:
Yeah I like that :) Or `InvalidUpateMemberValidity` but I think we are renaming  `update_member` to `update_investor` eventually I guess?